backend-api/src/functions/adapters/dynamoDbAdapter.ts

```diff
@@ -32,10 +55,83 @@ export class DynamoDbAdapter {
     }),
   });
 
-  constructor(tableName: string) {
+  constructor(
+    tableName: string,
+    private readonly getItemToSessionConvertor = getDynamoDBItemToSessionConvertor,
+  ) {
     this.tableName = tableName;
   }
 
+  async getSessionWithState(
+    sessionId: string,
+    sessionState: SessionState.AUTH_SESSION_CREATED,
+  ): Promise<Result<AuthSessionCreatedSession, ReadSessionError>>;
+  async getSessionWithState(
+    sessionId: string,
+    sessionState: SessionState.BIOMETRIC_TOKEN_ISSUED,
+  ): Promise<Result<BiometricTokenIssuedSession, ReadSessionError>>;
+  async getSessionWithState(
+    sessionId: string,
+    sessionState: SessionState.BIOMETRIC_SESSION_FINISHED,
+  ): Promise<Result<BiometricSessionFinishedSession, ReadSessionError>>;
+  async getSessionWithState(
+    sessionId: string,
+    sessionState: SessionState.RESULT_SENT,
+  ): Promise<Result<BiometricSessionFinishedSession, ReadSessionError>>;
+  async getSessionWithState(
+    sessionId: string,
+    sessionState: SessionState,
+  ): Promise<Result<Session, ReadSessionError>> {
+    let output: GetItemCommandOutput;
+    try {
+      console.log("Get session attempt"); // replace with proper logging
```

Comment:
In the past we only log by exception (apart from started, completed logs), do you think here there's a good reason to change this?

Reply:
See the PR description (stray comments) for the reasoning here. tl;dr, yes, I think it's very useful for working with external calls. But not crucial to this POC, definitely something to discuss in more detail elsewhere.

Reply:
Sorry should have re read that first, thanks for confirming!